python/pyarrow/tests/test_scalars.py

```diff
@@ -514,7 +515,7 @@ def test_struct():
         s['non-existent']
 
     s = pa.scalar(None, type=ty)
-    assert list(s) == []
+    assert list(s) == list(s.keys()) == ['x', 'y']
```

Comment:
This is a change in behaviour, but it seems more correct to me, since you can still do `s['x']` (which gives you a null scalar). But on the other hand it conflicts with `s.as_py()` being `None` (and not `{'x': None, 'y': None}`)

Reply:
Yes, it's a bit difficult. @wesm @xhochy  What do you think?

Reply:
I have no deep experience with scalars so I cannot contribute an opinion here.